server/src/main/java/org/elasticsearch/ingest/PipelineGraphStructureException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
+/**
+ * This exception is thrown when there is something wrong with the structure of the graph of pipelines to be applied to a document.
+ * For example, this is thrown when there are cycles in the graph.
+ */
+public class PipelineGraphStructureException extends ElasticsearchException {
```

Comment:
Should we make this more generic and name it `GraphStructureException`, so that in the future if we have another thing like this (say, for Painless), then we can re-use this one?